### PR TITLE
Load config from s3 (instead of environment variables)

### PR DESCRIPTION
**Pull request by jacobwinch**
Using environment variables for this project is starting to get a bit clunky (adding the Salesforce case-raising logic introduced a number of secret variables), so this PR re-factors to read in a JSON file from s3 instead.

This is similar to the approach taken in the fulfilment-lambdas project, which seems to work well there.

cc @AWare @johnduffell @paulbrown1982 @pvighi @lmath 

Before merge:

- [x] Add PROD conf file to S3 
- [x] Cloudform PROD stack

After merge:
- [x] Remove environment variables from PROD

### CODE REVIEW

File: cloudformation.yaml [hunk 1]
```diff
@@ -32,6 +32,12 @@ Resources:
                 - logs:PutLogEvents
                 - lambda:InvokeFunction
               Resource: "*"
+        - PolicyName: ReadPrivateCredentials
+          PolicyDocument:
+            Statement:
+              Effect: Allow
+              Action: s3:GetObject
+              Resource: !Sub arn:aws:s3:::fulfilment-lookup-private/${Stage}/*
```
Comment (johnduffell) [inline cloudformation.yaml:40]: I'm not sure if we should share membership-private and use a separate dir, or a new bucket (I had this issue with the now defunct supporter-metrics-job lambda which I should probably clean up from AWS)

File: src/main/scala/com/gu/fulfilmentLookup/Lambda.scala [hunk 2]
```diff
@@ -6,27 +6,37 @@ import java.time.format.DateTimeFormatter
...
             logger.info(s"Received request the following data: ${validLookup.value}")
-            val response = lookUp(validLookup.value, outputStream)
-            outputForAPIGateway(outputStream, Json.toJson(response))
+            loadConfig match {
+              case Success(config) => {
+                val response = lookUp(config, validLookup.value, outputStream)
+                outputForAPIGateway(outputStream, Json.toJson(response))
+              }
+              case Failure(error) => {
+                logger.error(s"Failed to load config in $stage due to $error")
```
Comment (johnduffell) [inline src/main/scala/com/gu/fulfilmentLookup/Lambda.scala:35]: will we get a sentry error if this happens?

File: src/main/scala/com/gu/fulfilmentLookup/Lambda.scala [hunk 3]
```diff
@@ -6,27 +6,37 @@ import java.time.format.DateTimeFormatter
...
 import com.gu.fulfilmentLookup.ResponseWriters._
-import scala.util.{ Failure, Success }
+import scala.util.{ Failure, Success, Try }
 import scalaz.{ -\/, \/- }
 
 trait FulfilmentLookupLambda extends Logging {
 
-  def s3Client: CsvClient
+  def csvClient: CsvClient
   def caseService: CaseService
```
Comment (johnduffell) [inline src/main/scala/com/gu/fulfilmentLookup/Lambda.scala]: not part of this PR, but caseService you only use raiseCase, why not pass in the function instead of the whole service? or rename the class to RaiseCase if you want a simpler type.  If we wanted a ListCases it probably wouldn't make sense that any code that needs to Raise cases would also get that code as a dependency.

**Comment by johnduffell** [comment]
great idea, save on parameters in CF!  I've taken the chance to comment on some other stuff, but don't let it stop this going out 👍

File: cloudformation.yaml (hunk 1 above)
Comment (jacobwinch) [reply cloudformation.yaml:40]: Yes, I opted for following current convention after some deliberation... however, current convention does produce a lot of tiny S3 buckets - we have a similar issue with everything having it's own dist bucket! Perhaps we need a membership AWS account spring clean...

Comment (jacobwinch) [reply cloudformation.yaml:40]: Ah, I see now that there isn't even a consistent convention

File: src/main/scala/com/gu/fulfilmentLookup/Lambda.scala (hunk 2 above)
Comment (jacobwinch) [reply src/main/scala/com/gu/fulfilmentLookup/Lambda.scala:35]: Not at the moment... all of our lambda-based projects (including fulfilment-lambdas, I think?) need to be configured with Sentry/Kibana and have a more general monitoring strategy implemented

**Comment by pvighi** [comment]
looks good  👍  

Not related to this PR but:

Remember we are fulfiling from the fulfilment_lambdas file now and we should expect a little more differences since me moved the trigger to 7 AM GMT.

For example today there is an extra sub (cancelled after 7 GMT) and a missing sub (cancelled sub on its last day) on the real file compared to the file you are using here.

**Comment by jacobwinch** [comment]
Thanks @pvighi - I'm hoping that we merge https://github.com/guardian/fulfilment-lambdas/pull/74 soon so that I only need to tweak the file/bucket once... if that doesn't happen tomorrow morning then I'll make the change anyway.

File: cloudformation.yaml (hunk 1 above)
Comment (johnduffell) [reply cloudformation.yaml:40]: yep agreed about the dist buckets!  maybe we just need a general labmdas-private and lambdas-dist

File: src/main/scala/com/gu/fulfilmentLookup/SalesforceCaseService.scala [hunk 4]
```diff
@@ -73,7 +54,31 @@ object SalesforceCaseService extends CaseService with Logging {
     }
   }
 
-  override def raiseCase(config: Config, lookupRequest: LookupRequest, lookupResponseBody: LookupResponseBody): String \/ Boolean = {
+}
+
+trait RaiseCase {
+  def open(config: Config, lookupRequest: LookupRequest, lookupResponseBody: LookupResponseBody): String \/ Boolean
```
Comment (johnduffell) [inline src/main/scala/com/gu/fulfilmentLookup/SalesforceCaseService.scala:60]: if you call it "apply" then you can use it with RaiseCase(...) rather than RaiseCase.open(...)

it's a bit magic though 🎉 , but it can be useful - your choice!

**Comment by jacobwinch** [comment]
Tested with my own SF contact in PROD and it's working correctly 🎉
